### DISCUSSION TITLE
Controlling output formatting

### DISCUSSION THREAD

**Discussion by noelwelsh** [planet42/Laika#373] [Q&A] [answered]
I'm trying to style sections that consist of a title and then a varying amount of text, code, and possibly other content. I have many of these types of sections; they are exercise solutions. My desired styling is to hide the body content until the user clicks an UI element to reveal it. At the moment I only care about HTML output. 

I could use the `detail` HTML tag, but there is no way I can see in `RawContent` to embed AST nodes, and I cannot see a way to get a formatter to convert the AST nodes to a `String`.

I could easily write a snippet of Javascript to do the styling, but I can't see how to get the HTML output I need. I thought I could use a `Section` but it doesn't wrap a `div` around the section, so I don't know where the solution content ends. Additionally, if I call `withStyle` I get an `id` not a `class` on the generated output (?!?!?!?!)

Any suggestions to solve this problem would be great.

**Comment by jenshalm** [comment, marked as answer]
The best approach depends a bit on your context. Here are some options:

* If you are working with Markdown sources, the `@:style` directive might be the simplest option (https://planet42.github.io/Laika/latest/07-reference/01-standard-directives.html#style). You can wrap any number of block elements in this directive. Those would then be wrapped in a `div` with a class attribute matching your directive attribute.
* If you don't want a `div` the simplest path might still be using the `@:style` directive and then using a customized renderer for this node type (by matching on a `BlockSequence` that has the style you assigned).
* If you want/need to work with the AST directly, the `withStyle` method should actually work as expected (see comment below).

> I could use the detail HTML tag, but there is no way I can see in RawContent to embed AST nodes, and I cannot see a way to get a formatter to convert the AST nodes to a String.

I cannot comment on this as I am not sure why you wanted to use a `RawContent` node with embedded AST here (meaning I am not sure what exactly you tried to do).

> Additionally, if I call withStyle I get an id not a class on the generated output (?!?!?!?!)

This would be a bug. However, the implementation and all the tests using `withStyle` look ok. To investigate it would be nice to see a minimal example to reproduce the behaviour, ideally in a separate ticket.

**Comment by noelwelsh** [comment]
Thanks for the advice! I've been able to solve the problem.
